content/blog/its-time-to-try-out-pixi/contents.lr

```diff
@@ -0,0 +1,1040 @@
+saying that the lock files are out of date,
+**a `pixi install` + `git commit` and `git push` will do the trick.**
+
+### CUDA specification syntax
+
+The syntax for specifying CUDA packages from `conda-forge`,
+is also something we need to commit to memory.
+Some patterns might look like:
+
+```toml
+package_name = { version = "some version", build = "*cuda*" }
+package_name = { version = "some version", build = "*cuda12*" }
+```
+
+### Default channels
+
+The default pixi behaviour is to look for packages from `conda-forge`.
+If your Python package can't be resolved from there,
+try moving it to the `pypi-dependencies` section instead.
+
+### Global tool installation
+
+If you went cold turkey as I did but still need to install Python tooling globally,
+then there's muscle memory that you'll need to un-learn very quickly.
+Whereas previously I could rely on a base environment
+through my `mambaforge` installation, it no longer exists.
+Rather, I needed to install Python and `pipx` globally with `pixi`
+and then use `pipx` (not `pixi`!) to install that tool using
+
+```bash
+pipx install --python $(which python) <tool name>
+```
+
+In this way, I could install `pyds-cli` and `llamabot` "globally"
+rather than within each environment.
+
+### Define `pixi` tasks thoughtfully
+
+Providing a task that allows for a one-command "get something useful accomplished"
+can be incredibly confidence-building for the next person
+who clones your repo locally and tries to run something.
+Examples might be a `lab` command to run JupyterLab,
+a `start` command that runs a Python script that demos the output of your work,
+or a `test` command to help software developers
+gain confidence that they've got the repo installed correctly.
+Be sure to provide that command in the README!
+
+### Minimal analyses
+
+What if you're not a tool developer
+and just want a quick environment to do some analyses
+without touching other environments?
+For this persona, there is a simplified workflow that I have used
+which may be helpful as a reference.
+
+Navigate to an empty directory.
+Then:
+
+```bash
+# Initialize pixi files
+pixi init
+# Add packages that you need
+pixi add jupyter ipython ipykernel pixi-kernel numpy pandas matplotlib seaborn scikit-learn
+# Run Jupyter lab
+pixi run jupyter lab
+# Add more packages as you see fit when it's needed; pixi will keep things in sync!
+pixi add statsmodels
+```
+
+Thanks to `pixi-kernel`,
+there will be a Jupyter kernel named `Pixi - Python 3 (ipykernel)`
+which will contain the packages contained in your environment.
+If you commit the notebooks + pixi configuration files to source control,
+someone else can download it and reproduce the environment easily.
+And as long as you don't rely on data living at a hard-coded path,
+that other person should be able to reproduce your work as well!
+
+## Acknowledgments
+
+I'd like to thank Sean Law and Reuben Arts for reviewing this content,
```

Comment:
```suggestion
I'd like to thank Sean Law and Ruben Arts for reviewing this content,
```
I guess you meant me :wink:

Reply:
Yes, that is right, and thank you for the catch, @ruben-arts!